x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/restriction/WorkflowsRestriction.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.security.authz.restriction;
+
+import org.elasticsearch.xpack.core.security.support.StringMatcher;
+
+import java.util.Set;
+import java.util.function.Predicate;
+
+public final class WorkflowsRestriction {
+
+    /**
+     * Default behaviour is no restriction which allows all workflows.
+     */
+    public static final WorkflowsRestriction NONE = new WorkflowsRestriction(null);
+
+    private final Set<String> names;
+    private final Predicate<String> predicate;
+
+    public WorkflowsRestriction(Set<String> names) {
+        assert names == null || names.size() > 0 : "workflow names cannot be an empty set";
+        this.names = names;
+        if (names != null) {
+            this.predicate = StringMatcher.of(names);
+        } else {
+            this.predicate = name -> true;
+        }
+    }
+
+    public boolean hasWorkflows() {
+        return this.names != null && this.names.size() > 0;
+    }
+
+    public boolean isWorkflowAllowed(String workflow) {
+        return predicate.test(workflow);
```

Comment:
> can we remove StringMatcher and replace it with just Set::contains as the predicate

I've removed it. Makes more sense to use `Set::contains` since we don't support wildcards at all.